jobs/payment-jobs/tasks/bcol_refund_confirmation_task.py

```diff
@@ -112,7 +112,11 @@ def _compare_and_update_records(cls, invoice_refs: List[InvoiceReference], bcol_
                 current_app.logger.error('Invoice refund total mismatch for %s', invoice_ref.invoice_number)
                 continue
 
-            # refund was processed and value is correct. Update invoice state.
+            refund = Refund.find_by_invoice_id(invoice_ref.invoice_id)
+
+            # refund was processed and value is correct. Update invoice state and refund date
             invoice.invoice_status_code = InvoiceStatus.REFUNDED.value
+            if refund is not None:
+                invoice.refund_date = refund.requested_date
```

Comment:
hmm I think this is not correct. Shouldn't `invoice.refund_date` be set to now as that is when we are flipping the state to refunded?